### PR TITLE
Revert "Revert optimization removal (#1352)"

### PR DESCRIPTION
**Pull request by jimma**
This reverts commit 263e92e2da1a0e934510a61f7525b1f310d13924.

### CODE REVIEW

File: jaxb-ri/runtime/impl/pom.xml [hunk 1]
```diff
@@ -71,83 +71,6 @@
...
-                        </configuration>
-                    </execution>
-                </executions>
-            </plugin>
-            <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-compiler-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <id>default-compile-mr</id>
```
Comment (lukasj) [inline jaxb-ri/runtime/impl/pom.xml:100]: this execution need to stay

**Review by lukasj — COMMENTED** [review]
In general, this is a good start.
What needs to be done:
* the output of the build is supposed to be multi-release jar but the execution to compile versioned sources is being removed, which is wrong
* there are few other files to be removed in tests and in `src/main/java-mr` folder

File: jaxb-ri/runtime/impl/pom.xml (hunk 1 above)
Comment (jimma) [reply jaxb-ri/runtime/impl/pom.xml]: Fixed

**Comment by jimma** [comment]
@lukasj  Thanks for the review. I've fixed these things. Please review this again.

File: jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/runtime/FilterTransducer.java [hunk 2]
```diff
@@ -1,5 +1,9 @@
 /*
+<<<<<<< HEAD:jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/runtime/FilterTransducer.java
  * Copyright (c) 1997, 2020 Oracle and/or its affiliates. All rights reserved.
+=======
+ * Copyright (c) 1997, 2018 Oracle and/or its affiliates. All rights reserved.
+>>>>>>> parent of 263e92e (Revert optimization removal (#1352)):jaxb-ri/runtime/impl/src/main/java/com/sun/xml/bind/v2/runtime/FilterTransducer.java
```
Comment (lukasj) [inline jaxb-ri/runtime/impl/src/main/java/org/glassfish/jaxb/runtime/v2/runtime/FilterTransducer.java:6]: looks like wrong merge here

**Review by lukasj — COMMENTED** [review]
there are also few more things which can be removed - `*.bytecode` package, `AccessorInjector` at least

**Comment by lukasj** [comment]
I'll just fix/remove the rest myslef...

**Comment by jimma** [comment]
Thanks @lukasj. Are you going to back port this change to  2.3.x too ?

**Comment by lukasj** [comment]
it's already there, see #1540

**Comment by jimma** [comment]
@lukasj  Nice! Thanks.

**Pull request merged**